### PR TITLE
Move to https and fix broken links

### PR DESCRIPTION
**Pull request by williamdes**
openspf seems to have moved, I checked and the previous website is the same as the current working one: https://web.archive.org/web/20190225203208/http://www.openspf.org/

I moved manually all links to https if possible.
And used https://github.com/tcort/markdown-link-check `find . -name \*.md -print0 | xargs -0 -n1 markdown-link-check` to do the final fixes

### CODE REVIEW

File: performance.md [hunk 1]
```diff
@@ -29,10 +29,10 @@ Global optimizations are used to speed up the overall messages processing improv
...
 
 * Unlike SA, Rspamd uses **specific state machines** to parse email components: mime structure, HTML parts, URLs, images, received headers and so on and so forth. This approach allows to skip unnecessary details and extract information from emails quicker than by using a large set of regular expressions for these purposes.
 
-* **[Hyperscan](https://github.com/intel/hyperscan) engine** is used in Rspamd to quickly process large regular expressions set. Unlike traditional regexps engines, Hyperscan allows to process multiple expressions at the same time. There are many details about hyperscan that are covered in the following [slides](http://www.slideshare.net/VsevolodStakhov/rspamdhyperscan).
+* **[Hyperscan](https://github.com/intel/hyperscan) engine** is used in Rspamd to quickly process large regular expressions set. Unlike traditional regexps engines, Hyperscan allows to process multiple expressions at the same time. There are many details about hyperscan that are covered in the following [slides](https://www.slideshare.net/VsevolodStakhov/rspamdhyperscan).
```
Comment (moisseev) [inline performance.md:36]: This page is unavailable: https://www.slideshare.net/VsevolodStakhov/rspamdhyperscan

File: performance.md [hunk 2]
```diff
@@ -29,10 +29,10 @@ Global optimizations are used to speed up the overall messages processing improv
 
 Rspamd uses various methods to speed up each individual message processing stage. This is achieved by applying local optimizations techniques:
 
-* **<abbr title="Abstract Syntax Tree">AST</abbr> optimizations** are used to exclude unnecessary checks from rules. You can watch the following [slides](http://www.slideshare.net/VsevolodStakhov/astrspamd) to get more details about this method.
+* **<abbr title="Abstract Syntax Tree">AST</abbr> optimizations** are used to exclude unnecessary checks from rules. You can watch the following [slides](https://www.slideshare.net/VsevolodStakhov/astrspamd) to get more details about this method.
```
Comment (moisseev) [inline performance.md:32]: This page is unavailable: https://www.slideshare.net/VsevolodStakhov/astrspamd

File: _posts/2016-08-01-rspamd-1.3.1.md [hunk 3]
```diff
@@ -110,4 +110,4 @@ rule "local" {
 
 ### Other bugfixes and improvements
 
-There are couple of other important bugfixes and improvements including a critical fix for extracting values from the top received header. The test framework has been improved with new functional tests and better integration with [CircleCI](https://circleci.com/gh/vstakhov/rspamd).
\ No newline at end of file
+There are couple of other important bugfixes and improvements including a critical fix for extracting values from the top received header. The test framework has been improved with new functional tests and better integration with [DroneCI](https://ci.rspamd.com/rspamd/rspamd).
```
Comment (moisseev) [inline _posts/2016-08-01-rspamd-1.3.1.md:113]: I am not sure if we should rewrite the events of the past like this. It really was CircleCI in 2016.

File: performance.md (hunk 1 above)
Comment (williamdes) [reply performance.md:36]: I checked it and it is ;)
Here is the screenshot proof
Screenshot_2023-07-18-16-18-04-193_org mozilla fenix

File: performance.md (hunk 2 above)
Comment (williamdes) [reply performance.md:32]: Same as above, it's available

File: _posts/2016-08-01-rspamd-1.3.1.md (hunk 3 above)
Comment (williamdes) [reply _posts/2016-08-01-rspamd-1.3.1.md:113]: Ah yes okay, I just renamed it.
sorry I should have done something else
should I use the web archive maybe?

Comment (vstakhov) [reply _posts/2016-08-01-rspamd-1.3.1.md:113]: I don't have a strong opinion here: either option, removing a link or redirecting it to the actual CI is fine to me. I think it's easier to leave the change untouched, as it is still better than a broken link.

**Review by vstakhov — APPROVED** [review]

File: _posts/2016-08-01-rspamd-1.3.1.md (hunk 3 above)
Comment (williamdes) [inline _posts/2016-08-01-rspamd-1.3.1.md:113]:
```suggestion
There are couple of other important bugfixes and improvements including a critical fix for extracting values from the top received header. The test framework has been improved with new functional tests and better integration with CircleCI
```

**Pull request merged**